### PR TITLE
Tm bug fixes

### PR DESCRIPTION
**Pull request by JaseZiv**

### CODE REVIEW

File: R/tm_player_injury_history.R [hunk 1]
```diff
@@ -48,19 +48,20 @@ tm_player_injury_history <- function(player_urls) {
...
+                              error = function(e) injured_until <- NA_character_) %>% .replace_empty_na()
       duration <- tryCatch(pg %>% rvest::html_nodes(".zentriert+ td.rechts") %>% rvest::html_text(),
-                            error = function(e) duration <- NA_character_)
+                            error = function(e) duration <- NA_character_) %>% .replace_empty_na()
       games_missed <- tryCatch(pg %>% rvest::html_nodes(".wappen_verletzung") %>% rvest::html_text() %>% as.numeric() %>% suppressWarnings(),
-                              error = function(e) games_missed <- NA_integer_)
+                              error = function(e) games_missed <- NA_integer_) %>% .replace_empty_na()
       club <- tryCatch(pg %>% rvest::html_nodes("img") %>% rvest::html_attr("alt"),
-                        error = function(e) goals <- NA_character_)
+                        error = function(e) goals <- NA_character_) %>% .replace_empty_na()
```
Comment (JaseZiv) [inline R/tm_player_injury_history.R:63]: Using internal function `.replace_empty_na()` everywhere to ensure values return a NA and allow us to construct the final df output

File: R/tm_player_injury_history.R [hunk 2]
```diff
@@ -26,9 +26,9 @@ tm_player_injury_history <- function(player_urls) {
 
     injury_page <- xml2::read_html(player_url_fixed)
 
-    player_name <- injury_page %>% rvest::html_nodes("h1") %>% rvest::html_text()
+    player_name <- injury_page %>% rvest::html_nodes("h1") %>% rvest::html_text() %>% gsub("#[0-9]+ ", "", .) %>% stringr::str_squish()
 
-    injury_urls <- injury_page %>% html_nodes(".tm-pagination__list-item a") %>% html_attr("href")
+    injury_urls <- injury_page %>% rvest::html_nodes(".tm-pagination__list-item a") %>% rvest::html_attr("href")
```
Comment (JaseZiv) [inline R/tm_player_injury_history.R:31]: Drive by fix for player numbers also coming through with the player name

File: tests/testthat/test-transfermarkt.R [hunk 3]
```diff
@@ -158,14 +158,14 @@ test_that("tm_league_debutants() works", {
 test_that("tm_expiring_contracts() works", {
   testthat::skip_on_cran()
 
-  laliga_expiring <- tm_expiring_contracts(country_name = "Spain", contract_end_year = 2023)
+  laliga_expiring <- tm_expiring_contracts(country_name = "Spain", contract_end_year = 2024)
```
Comment (JaseZiv) [inline tests/testthat/test-transfermarkt.R:161]: This was failing as the 2023 season is officially over on transfermarkt

**Comment by JaseZiv** [comment]
Failed test is for `fotmob_get_match_players()`

**Review by tonyelhabr — APPROVED** [review]
lgtm

**Pull request merged**